### ISSUE TITLE
[NCC] Agent's Toolkit did not receive the correct extra counters on ETB with Winding Constrictor and Vorinclex, Monstrous Raider; also caused NPE when a creature entered the battlefield

### ISSUE THREAD

**Issue by Grath** [magefree/mage#9580]
I played an [[Agent's Toolkit]] with [[Winding Constrictor]] and [[Vorinclex, Monstrous Raider]] in play. It received four +1/+1 counters, but only one of each ability counter. It should have received four of each type of counter.

Then, I attacked with [[Sanctuary Warden]], drawing a card and creating a 1/1 Citizen token. The client showed a null pointer exception, but the server logs (playing against AIs) showed:
```
java.io.NotSerializableException: java.util.HashMap$KeySet
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at mage.remote.traffic.ZippedObjectImpl.zip(ZippedObjectImpl.java:31)
	at mage.remote.traffic.ZippedObjectImpl.<init>(ZippedObjectImpl.java:23)
	at mage.utils.CompressUtil.compress(CompressUtil.java:56)
	at mage.interfaces.callback.ClientCallback.setData(ClientCallback.java:57)
	at mage.interfaces.callback.ClientCallback.<init>(ClientCallback.java:26)
	at mage.interfaces.callback.ClientCallback.<init>(ClientCallback.java:20)
	at mage.server.game.GameSessionPlayer.lambda$chooseChoice$5(GameSessionPlayer.java:89)
	at java.util.Optional.ifPresent(Optional.java:159)
	at mage.server.game.GameSessionPlayer.chooseChoice(GameSessionPlayer.java:88)
	at mage.server.game.GameController.lambda$chooseChoice$17(GameController.java:825)
	at mage.server.game.GameController.perform(GameController.java:988)
	at mage.server.game.GameController.perform(GameController.java:981)
	at mage.server.game.GameController.chooseChoice(GameController.java:825)
	at mage.server.game.GameController.lambda$init$3aa0d15c$2(GameController.java:217)
	at mage.game.events.EventDispatcher.fireEvent(EventDispatcher.java:28)
	at mage.game.events.PlayerQueryEventSource.chooseChoice(PlayerQueryEventSource.java:92)
	at mage.game.GameImpl.fireChooseChoiceEvent(GameImpl.java:2798)
	at mage.player.human.HumanPlayer.choose(HumanPlayer.java:479)
	at mage.cards.a.AgentToolkitMoveCounterEffect.apply(AgentsToolkit.java:97)
	at mage.abilities.AbilityImpl.resolveMode(AbilityImpl.java:201)
	at mage.abilities.AbilityImpl.resolve(AbilityImpl.java:185)
	at mage.abilities.TriggeredAbilityImpl.resolve(TriggeredAbilityImpl.java:149)
	at mage.game.stack.StackAbility.resolve(StackAbility.java:89)
	at mage.game.GameImpl.resolve(GameImpl.java:1634)
	at mage.game.GameImpl.playPriority(GameImpl.java:1560)
	at mage.game.turn.Step.priority(Step.java:61)
	at mage.game.turn.Phase.playStep(Phase.java:184)
	at mage.game.turn.Phase.play(Phase.java:89)
	at mage.game.turn.Turn.play(Turn.java:132)
	at mage.game.GameImpl.playTurn(GameImpl.java:1055)
	at mage.game.GameImpl.play(GameImpl.java:965)
	at mage.game.GameImpl.start(GameImpl.java:941)
	at mage.server.game.GameWorker.call(GameWorker.java:34)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by Grath** [comment]
[Winding Constrictor]] [[Vorinclex, Monstrous Raider]] [[Sanctuary Warden]]
(The warden is probably irrelevant, it's just what caused a token to be created.)

**Comment by Grath** [comment]
[[Winding Constrictor]] stupid brackets

**Comment by Grath** [comment]
Worth noting, proliferate (from [[Contagion Engine]]) did correctly result in bonus counters of each kind on Agent's Toolkit, so this is a bug specifically in the handling of _enters the battlefield_ for Winding Constrictor and Vorinclex.

**Comment by Grath** [comment]
Agent's Toolkit error happened precisely the same the following turn when I played a creature card, so it's not token vs full creature.

**Comment by satchmosanzabar** [comment]
I just played Agent's Toolkit for the first time, and the trigger just does not work. It goes on the stack, but the client displays a pop-up with "null pointer exception".

**Comment by JayDi85** [comment]
Fixed by ad3b76dae6165f3becdf227a93fe6208b359af71
